code/__HELPERS/icons.dm

```diff
@@ -762,6 +762,23 @@ world
 	else
 		curdir = A.dir
 
+	//Let's check if the icon actually contains any diagonals, just skip if it's south to save (lot of) time
+	if(curdir != SOUTH)
+		var/icon/test_icon 
+		var/directionals_exist = FALSE

```

Comment:
You could probably get your processing back if you just cache the outcome of this and check that

Reply:
:+1: caching is something i've been experimenting with but instead i'd use icon_state()